src/auth.ts

```diff
@@ -1,26 +1,52 @@
-import * as fs from 'fs';
-import * as os from 'os';
 import * as path from 'path';
 import * as core from '@actions/core';
 import * as io from '@actions/io';
-import {create as xmlCreate} from 'xmlbuilder2';
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+import { create as xmlCreate } from 'xmlbuilder2';
 import * as constants from './constants';
+import * as gpg from './gpg';
 
 export const M2_DIR = '.m2';
 export const SETTINGS_FILE = 'settings.xml';
 
-export async function configAuthentication(
+export async function configureAuthentication() {
+  const id = core.getInput(constants.INPUT_SERVER_ID);
+  const username = core.getInput(constants.INPUT_SERVER_USERNAME);
+  const password = core.getInput(constants.INPUT_SERVER_PASSWORD);
+  const gpgPrivateKey =
+    core.getInput(constants.INPUT_GPG_PRIVATE_KEY) || constants.INPUT_DEFAULT_GPG_PRIVATE_KEY;
+  const gpgPassphrase =
+    core.getInput(constants.INPUT_GPG_PASSPHRASE) ||
+    (gpgPrivateKey ? constants.INPUT_DEFAULT_GPG_PASSPHRASE : undefined);
+
+  if (gpgPrivateKey) {
+    core.setSecret(gpgPrivateKey);
+  }
+
+  await createAuthenticationSettings(id, username, password, gpgPassphrase);
+
+  if (gpgPrivateKey) {
+    core.info('Importing private gpg key');
+    const keyFingerprint = (await gpg.importKey(gpgPrivateKey)) || '';
+    core.saveState(constants.STATE_GPG_PRIVATE_KEY_FINGERPRINT, keyFingerprint);
+  }
+}
+
+export async function createAuthenticationSettings(
   id: string,
   username: string,
   password: string,
   gpgPassphrase: string | undefined = undefined
 ) {
-  console.log(
-    `creating ${SETTINGS_FILE} with server-id: ${id};`,
-    'environment variables:',
-    `username=\$${username},`,
-    `password=\$${password},`,
-    `and gpg-passphrase=${gpgPassphrase ? '$' + gpgPassphrase : null}`
+  core.info(
+    `Creating ${SETTINGS_FILE} with server-id: ${id};
+     environment variables:
+     username=\$${username},
+     password=\$${password},
```

Comment:
Why are these secrets logged? I know GitHub Actions masks repo/org secrets in the output, but it's a really bad idea to rely on this and to log credentials in general.

Reply:
I don't have context why this logging was initially added in V1 but actually, it doesn't even log secrets - it logs names of variables where these values are stored. 
See action usage: https://github.com/actions/setup-java/pull/134/files#diff-e8f077830705686ec3300dbcec8ee6ac6345ae9a2965508ff3abc4146d520bf8R154
User doesn't pass secrets to action, it passes names of variables where secrets are located.

Anyway, I agree that it doesn't make much sense to log this info into logs. I have cut the log a bit